python_modules/libraries/dagster-aws/dagster_aws/pipes.py

```diff
@@ -174,13 +198,53 @@ def read_messages(
             self._handler = None
 
     def consume_cloudwatch_logs(
-        self, client: boto3.client, log_group: str, log_stream: str
+        self,
+        log_group: str,
+        log_stream: str,
+        start_time: Optional[int] = None,
+        end_time: Optional[int] = None,
     ) -> None:
-        raise NotImplementedError("CloudWatch logs are not yet supported in the pipes protocol.")
+        handler = check.not_none(
+            self._handler, "Can only consume logs within context manager scope."
+        )
+
+        for events_batch in self._get_all_cloudwatch_events(
+            log_group=log_group, log_stream=log_stream, start_time=start_time, end_time=end_time
+        ):
+            for event in events_batch:
+                for log_line in event["message"].splitlines():
+                    extract_message_or_forward_to_stdout(handler, log_line)
```

Comment:
Ok so this means we will ingest all of stdout from cloudwatch. What about stderr?

Reply:
Funny enough, that's not how AWS Glue handles stdout/err. 

I haven't completely figured it out yet, but it seems like it actually routes stderr to `/out` LogGroup. Here is an example with some Dagster logs written by the default MessageWriter - that's stderr right? 

![image](https://github.com/user-attachments/assets/916666b3-aff1-4cec-88ab-2bd24fb14c6f)

Meanwhile, `/error` LogGroup contains multiple streams: spark worker logs and the mains spark process logs. That's how it looks like: 

![image](https://github.com/user-attachments/assets/c919c52d-cd78-42bc-bf89-5ec43a2d0de6)

No trace of Python/Dagster stderr.